internal/dag/status.go

```diff
@@ -67,49 +67,50 @@ func (sw *StatusWriter) commit(osw *ObjectStatusWriter) {
 		Name:      osw.obj.GetObjectMeta().GetName(),
 		Namespace: osw.obj.GetObjectMeta().GetNamespace(),
 	}
-	if _, ok := sw.statuses[m]; !ok {
+	if _, ok := sw.proxyStatuses[m]; !ok {
 		// only record the first status event
-		sw.statuses[m] = Status{
+		sw.proxyStatuses[m] = ProxyStatusUpdate{
 			Object:      osw.obj,
 			Status:      osw.values["status"],
 			Description: osw.values["description"],
 			Vhost:       osw.values["vhost"],
 		}
 	}
 }
-func (osw *ObjectStatusWriter) WithValue(key, val string) *ObjectStatusWriter {
-	osw.values[key] = val
-	return osw
+
+func (psw *ProxyStatusWriter) WithValue(key, val string) *ProxyStatusWriter {
+	psw.values[key] = val
+	return psw
 }
 
-func (osw *ObjectStatusWriter) SetInvalid(format string, args ...interface{}) {
-	osw.WithValue("description", fmt.Sprintf(format, args...)).WithValue("status", k8s.StatusInvalid)
+func (psw *ProxyStatusWriter) AddInvalidCondition(conditionType, formattedReason string, args ...interface{}) {
```

Comment:
Need some comment or example to explain what this will eventually do with a `DetailedCondition`

Reply:
It will eventually be used to construct a SubCondition under the `Valid` DetailedCondition.